using the lua script with SITL

vzarkar opened the pull request:
Updated description - Lua script with SITL

Hwurzburg commented on libraries/AP_Scripting/examples/mission-load.lua:
```diff
@@ -1,6 +1,8 @@
 -- Example of loading a mission from the SD card using Scripting
 -- Would be trivial to select a mission based on scripting params or RC switch
 -- luacheck: only 0
+--In case of testing it with SITL on linux - copy this "mission-load.lua" script to "ardupilot/ArduCopter/scripts" directory
```
I dont have any issue adding this comment in the lua script once its been corrected, but its not necessary since we have an entire wiki page for running scripting in SITL in the wiki

vzarkar replied:
It is not about where the lua script should be. It is mentioned in the wiki page. But where the mission1.txt file should be is not clear from wiki. If anyone is using SITL or using MAVFTP to transfer file to sdcard... the directory where to copy this waypoint file(mission1.txt) needs to be mentioned. It took me hours to figure it out. On drone with Orange cube where you have no direct access to sdcard, I couldn't easily find where to copy this mission file using MAVFTP.

magicrub replied:
then the text should say something like "put the mission1.txt file in the same directory as the script". Being absolute with the directory name makes this unnecessarily complicated

vzarkar replied:
Place "mission1.txt" in the directory one level above the "scripts" directory in case of SITL. In case of flight controller, it should be put at level as APM.
- APM
  - scripts
- mission1.txt

Hwurzburg replied:
I do no think this is correct..the lua io.open call expects the file (unless pathname is specified) to be from the same directory the script is called

vzarkar replied:
I tried and tested it and found the working solution that is mentioned in my previous message.It loads mission.txt file successfully from the location that I mentioned.

Hwurzburg commented on libraries/AP_Scripting/examples/mission-load.lua:
```diff
@@ -1,6 +1,8 @@
 -- Example of loading a mission from the SD card using Scripting
 -- Would be trivial to select a mission based on scripting params or RC switch
 -- luacheck: only 0
+--In case of testing it with SITL on linux - copy this "mission-load.lua" script to "ardupilot/ArduCopter/scripts" directory
```
yes, you are correct...
```suggestion
--Copy this "mission-load.lua" script to the "scripts" directory of the simulation or autpilot's SD card. The "mission1.txt" file containing the mission items should be placed in the directory above the "scripts" directory.
```

vzarkar replied:
Thanks for considering my request. In case of placing it on sdcard it requires mission item at the same level as APM directory other wise it will not work if it is inside APM directory. This is tested and confirmed with Orange blue flight controller.

Hwurzburg replied:
which is exactly what my suggested wording states

timtuxworth replied:
Well to be fair @Hwurzburg some people will still find this confusing. 
Would it not make it absolutely clear to add "In case of placing it on SD Card, mission1.txt file should be placed in the APM directory root."

Hwurzburg replied:
@timtuxworth that could be added, but cant replace anything in the suggested text since it applies to SITL, SD,......and ROMFS

timtuxworth replied:
Yes I'm suggesting to add it to what you proposed.

Hwurzburg commented on libraries/AP_Scripting/examples/mission-load.lua:
```diff
@@ -1,6 +1,8 @@
 -- Would be trivial to select a mission based on scripting params or RC switch
 -- luacheck: only 0
+--In case of testing it with SITL on linux - copy this "mission-load.lua" script to "ardupilot/ArduCopter/scripts" directory
+--and copy the "mission1.txt" file which contains waypoints to "ardupilot/ArduCopter" directory.
```
remove

Hwurzburg commented:
@vzarkar I have squashed this and added a proper commit message and tagged it for the next devcall

vzarkar commented:
Thanks.
________________________________
From: Henry Wurzburg ***@***.***>
Sent: Saturday, October 28, 2023 5:39 PM
To: ArduPilot/ardupilot ***@***.***>
Cc: vzarkar ***@***.***>; Mention ***@***.***>
Subject: Re: [ArduPilot/ardupilot] using the lua script with SITL (PR #25342)


@vzarkar<https://github.com/vzarkar> I have squashed this and added a proper commit message and tagged it for the next devcall

—
Reply to this email directly, view it on GitHub<https://github.com/ArduPilot/ardupilot/pull/25342#issuecomment-1783792110>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ACYA6BCDSVDEXQBMGBZLI43YBTYWLAVCNFSM6AAAAAA6L3XU6SVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTOOBTG44TEMJRGA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>
